2009-07-16  Ulrich Drepper  <drepper@redhat.com>

	[BZ #10360]
	* resolv/res-mkquery.c (__res_nopt): If anslen is > 0xffff store
	0xffff in the EDNS0 record.

diff --git a/resolv/res_mkquery.c b/resolv/res_mkquery.c
index ae0cdb4..2dda4c0 100644
--- a/resolv/res_mkquery.c
+++ b/resolv/res_mkquery.c
@@ -244,7 +244,7 @@ __res_nopt(res_state statp,
 	*cp++ = 0;	/* "." */
 
 	NS_PUT16(T_OPT, cp);	/* TYPE */
-	NS_PUT16(anslen & 0xffff, cp);	/* CLASS = UDP payload size */
+	NS_PUT16(MIN(anslen, 0xffff), cp);	/* CLASS = UDP payload size */
 	*cp++ = NOERROR;	/* extended RCODE */
 	*cp++ = 0;		/* EDNS version */
 	/* XXX Once we support DNSSEC we change the flag value here.  */
